lib/logstash/filters/mutate.rb

```diff
@@ -338,85 +334,91 @@ def gsub_dynamic_fields(event, original, needle, replacement)
     end
   end
 
-  private
   def uppercase(event)
     @uppercase.each do |field|
-      if event[field].is_a?(Array)
-        event[field].each { |v| v.upcase! }
-      elsif event[field].is_a?(String)
-        event[field].upcase!
-      else
-        @logger.debug("Can't uppercase something that isn't a string",
-                      :field => field, :value => event[field])
-      end
+      original = event[field]
+      event[field] = case original
+        when Array
+          original.map(&:upcase!)
```

Comment:
Tests still pass if I remove the `!` part of the method call:

Here's my patch:

```diff
diff --git a/lib/logstash/filters/mutate.rb b/lib/logstash/filters/mutate.rb
index d279525..367d5bb 100644
--- a/lib/logstash/filters/mutate.rb
+++ b/lib/logstash/filters/mutate.rb
@@ -339,9 +339,9 @@ module LogStash module Filters class Mutate < LogStash::Filters::Base
       original = event[field]
       event[field] = case original
         when Array
-          original.map(&:upcase!)
+          original.map(&:upcase)
         when String
-          original.upcase!
+          original.upcase
         else
           @logger.debug("Can't uppercase something that isn't a string",
                         :field => field, :value => original)
@@ -355,9 +355,9 @@ module LogStash module Filters class Mutate < LogStash::Filters::Base
       original = event[field]
       event[field] = case original
         when Array
-          original.map(&:downcase!)
+          original.map(&:downcase)
         when String
-          original.downcase!
+          original.downcase
         else
           @logger.debug("Can't lowercase something that isn't a string",
                         :field => field, :value => original)
```